staging/src/k8s.io/kube-aggregator/pkg/controllers/openapi/aggregator/downloader.go

```diff
@@ -95,12 +97,89 @@ func (s *Downloader) Download(handler http.Handler, etag string) (returnSpec *sp
 				}
 			}
 		}
+
 		return openAPISpec, newEtag, http.StatusOK, nil
 	default:
 		return nil, "", 0, fmt.Errorf("failed to retrieve openAPI spec, http error: %s", writer.String())
 	}
 }
 
+type gvList struct {
+	Paths []string `json:"Paths"`
+}
+
+func (s *Downloader) DownloadV3(handler http.Handler, etag string) (returnSpec map[string]*spec3.OpenAPI, err error) {
```

Comment:
I didn't expect a downloader at all. We moved to group-version instead of group alone in order to get rid of all this logic.

Reply:
https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/2896-openapi-v3/README.md#aggregator

This was the mechanism I described in the KEP. It's possible we can optimize it to avoid the download but I think we should resolve https://github.com/kubernetes/enhancements/pull/2898/files#r745380166 first.

Reply:
Per the discussion here and in https://github.com/kubernetes/enhancements/pull/2898/files#r745380166, I think proxying the requests instead of downloading is something that would be beneficial for performance. I think if we don't publish the `apis/g/v/openapi` route, we can still publish at `openapi/v3/apis/g/v` and bake a bit more proxy logic into the aggregator to avoid the download. 

If that's the path we'd like to take, would it be alright for this to be deferred to beta? This change should be 100% transparent to the user and is backwards compatible.